test/client/spec/victory-core/victory-util/scale.spec.js

```diff
@@ -79,7 +79,8 @@ describe("victory-util/scale", () => {
       expect(scaleType).to.equal("linear");
     });
 
-    it("uses data to distinguish between time and linear scales", () => {
+    // TODO: HERE -- FIGURE OUT WHY THIS IS FAILING.
+    it.skip("uses data to distinguish between time and linear scales", () => {
```

Comment:
@becca-bailey if you could re-enable this and the other `it.skip()` tests and figure out that would be great!

Reply:
Update: This is a real test failure. Something is likely wrong with `getType()` which tries inference based on internal knowledge I think of d3-scale that may have changed: https://github.com/FormidableLabs/victory/blob/main/packages/victory-core/src/victory-util/scale.js#L113-L128

Reply:
Here's some metadata about the shape of functions / objects in d3-scale that we use in `getTypes()`:

```js
{
  "scaleBand": {
    "type": "function",
    "keys": [
      "domain",
      "range",
      "copy",
      "rangeRound",
      "bandwidth",
      "step",
      "round",
      "padding",
      "paddingInner",
      "paddingOuter",
      "align"
    ]
  },
  "scaleDiverging": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleDivergingLog": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "base",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleDivergingPow": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "exponent",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleDivergingSqrt": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "exponent",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleDivergingSymlog": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "constant",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleIdentity": {
    "type": "function",
    "keys": [
      "invert",
      "range",
      "domain",
      "unknown",
      "copy",
      "ticks",
      "tickFormat",
      "nice"
    ]
  },
  "scaleImplicit": {
    "type": "object",
    "keys": []
  },
  "scaleLinear": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "clamp",
      "interpolate",
      "unknown",
      "copy",
      "ticks",
      "tickFormat",
      "nice"
    ]
  },
  "scaleLog": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "clamp",
      "interpolate",
      "unknown",
      "base",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleOrdinal": {
    "type": "function",
    "keys": [
      "domain",
      "range",
      "unknown",
      "copy"
    ]
  },
  "scalePoint": {
    "type": "function",
    "keys": [
      "domain",
      "range",
      "copy",
      "rangeRound",
      "bandwidth",
      "step",
      "round",
      "padding",
      "align"
    ]
  },
  "scalePow": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "clamp",
      "interpolate",
      "unknown",
      "exponent",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleQuantile": {
    "type": "function",
    "keys": [
      "invertExtent",
      "domain",
      "range",
      "unknown",
      "quantiles",
      "copy"
    ]
  },
  "scaleQuantize": {
    "type": "function",
    "keys": [
      "domain",
      "range",
      "invertExtent",
      "unknown",
      "thresholds",
      "copy",
      "ticks",
      "tickFormat",
      "nice"
    ]
  },
  "scaleRadial": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "round",
      "clamp",
      "unknown",
      "copy",
      "ticks",
      "tickFormat",
      "nice"
    ]
  },
  "scaleSequential": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleSequentialLog": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "base",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleSequentialPow": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "exponent",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleSequentialQuantile": {
    "type": "function",
    "keys": [
      "domain",
      "interpolator",
      "range",
      "quantiles",
      "copy"
    ]
  },
  "scaleSequentialSqrt": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "exponent",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleSequentialSymlog": {
    "type": "function",
    "keys": [
      "domain",
      "clamp",
      "interpolator",
      "range",
      "rangeRound",
      "unknown",
      "constant",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleSqrt": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "clamp",
      "interpolate",
      "unknown",
      "exponent",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleSymlog": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "clamp",
      "interpolate",
      "unknown",
      "constant",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleThreshold": {
    "type": "function",
    "keys": [
      "domain",
      "range",
      "invertExtent",
      "unknown",
      "copy"
    ]
  },
  "scaleTime": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "clamp",
      "interpolate",
      "unknown",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "scaleUtc": {
    "type": "function",
    "keys": [
      "invert",
      "domain",
      "range",
      "rangeRound",
      "clamp",
      "interpolate",
      "unknown",
      "ticks",
      "tickFormat",
      "nice",
      "copy"
    ]
  },
  "tickFormat": {
    "type": "function",
    "keys": [
      "toString"
    ]
  }
}
```

Reply:
And more specifically, the test is failing because `scaleLinear().unknown` is defined which causes `getType()` to incorrectly infer that `ordinal` is the scale type.